packages/ts/src/rules/octalEscapes.ts

```diff
@@ -0,0 +1,73 @@
+import * as ts from "typescript";
+
+import { typescriptLanguage } from "../language.js";
+
+/**
+ * Finds the position and length of an octal escape sequence in a string.
+ * Returns null if no octal escape is found.
+ */
+function findOctalEscape(
+	text: string,
+): null | { index: number; length: number } {
+	// Remove quotes from the string literal
+	const content = text.slice(1, -1);
+
+	// Match octal escapes: \0 followed by [0-7], or \1-7 optionally followed by [0-7]
+	// But exclude escaped backslashes (\\)
+	const octalEscapePattern = /(?<!\\)\\0[0-7]|(?<!\\)\\[1-7][0-7]*/;
+	const match = octalEscapePattern.exec(content);
+
+	if (!match || match.index === undefined) {
+		return null;
+	}
+
+	// Add 1 to account for the opening quote
+	return {
+		index: match.index + 1,
+		length: match[0].length,
+	};
+}
+
+export default typescriptLanguage.createRule({
+	about: {
+		description: "Reports using octal escape sequences in string literals.",
+		id: "octalEscapes",
+		preset: "logical",
+	},
+	messages: {
+		noOctalEscape: {
+			primary: "Octal escape sequences should not be used in string literals.",
```

Comment:
[Docs] Rule report messages should not be prescriptive "don't X". Instead, lean towards "do X". For example, instead of messages like "Octal escape sequences should not be used in string literals.", use messages like "Prefer hexadecimal or Unicode escape sequences over legacy octal escape sequences.".

Also make sure the `.mdx` docs start like this too.